src/platform/datapath_raw_xdp.c

```diff
@@ -809,7 +810,7 @@ CxPlatDpRawInterfaceAddRule(
     CxPlatLockAcquire(&Interface->RuleLock);
     // TODO - Don't always allocate a new array?
 
-    if (Interface->RuleCount + 1 == 0) {
+    if ((uint32_t)Interface->RuleCount + (uint32_t)Count > UINT8_MAX) {
```

Comment:
oh, we only support 255 connections for now?

Reply:
255 rules, which is based on the number of sockets, not connections. A single socket on the server can have nearly unlimited connections. Depending on the sharing configured on the client, it can too.